components/nexus-core/src/main/java/org/sonatype/nexus/proxy/maven/ChecksumContentValidator.java

```diff
@@ -194,37 +184,32 @@ private static RemoteHashResponse doRetrieveChecksumItem(ProxyRepository proxy,
         throw new LocalStorageException("Null item repository attributes");
       }
 
-      if (Boolean.parseBoolean(attributes.get(noattrname)) && !request.isRequestAsExpired()) {
-        throw new ItemNotFoundException(request);
+      // Check if checksum path is in NFC
+      if (proxy.getNotFoundCache().contains(checksumRequest.getRequestPath()) && !checksumRequest.isRequestAsExpired()) {
+        throw new ItemNotFoundException(checksumRequest);
       }
 
+      // If attributes does not contain checksum hash, then attempt to fetch and store it
       String hash = attributes.get(attrname);
-      if (hash == null || request.isRequestAsExpired()) {
+      if (hash == null || checksumRequest.isRequestAsExpired()) {
         try {
-          final StorageFileItem remoteItem =
-              (StorageFileItem) proxy.getRemoteStorage().retrieveItem(proxy, request, proxy.getRemoteUrl());
-          hash = MUtils.readDigestFromFileItem(remoteItem); // closes http input stream
-        }
-        catch (ItemNotFoundException e) {
-          // fall through
-        }
-        catch (RemoteAccessException e) {
-          // fall through
+          StorageFileItem checksumItem =
+              (StorageFileItem) proxy.getRemoteStorage().retrieveItem(proxy, checksumRequest, proxy.getRemoteUrl());
+
+          // fetched checksum item, extract hash and store it in attributes
+          hash = doStoreChechsumItem(proxy, artifact, attrname, checksumItem);
```

Comment:
Probably needs a `proxy.removeFromNotFoundCache(checksumRequest)` to evict it from NFC